op-e2e/blobs/blob_test.go

```diff
@@ -0,0 +1,160 @@
+package blobs
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"math/big"
+	mrand "math/rand"
+	"testing"
+	"time"
+
+	op_e2e "github.com/ethereum-optimism/optimism/op-e2e"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
+	"github.com/ethereum-optimism/optimism/op-e2e/faultproofs"
+	"github.com/ethereum-optimism/optimism/op-e2e/system/e2esys"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/testutils"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/consensus/misc/eip4844"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethstorage/da-server/pkg/da"
+	"github.com/ethstorage/da-server/pkg/da/client"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	dacPort = 37777
+)
+
+var (
+	ctx       = context.Background()
+	dacUrl    = fmt.Sprintf("http://127.0.0.1:%d", dacPort)
+	toAddress = testutils.RandomAddress(mrand.New(mrand.NewSource(dacPort)))
+	blobs     = make([]*eth.Blob, 3)
+)
+
+func TestFunctionSuccess(t *testing.T) {
+	op_e2e.InitParallel(t)
+	dacServer := StartDACServer(t)
+	defer dacServer.Stop(ctx)
+
+	sys, _ := faultproofs.StartFaultDisputeSystem(t, faultproofs.WithDAC(dacUrl))
+	t.Cleanup(sys.Close)
+	l2Client := sys.NodeClient(e2esys.RoleSeq)
+
+	for i := range blobs {
+		b := GetRandBlob(t, int64(i))
+		blobs[i] = &b
+	}
+
+	tx, err := sendTransactionWithBlobs(t, ctx, l2Client, sys.TestAccount(0), toAddress, blobs)
+	require.NoError(t, err)
+	_, err = wait.ForReceiptOK(ctx, l2Client, tx.Hash())
+
+	// TODO: wait for DAC server bug fix check in to refactor here
```

Comment:
Bug to prevent blobs from being downloaded in batch?

Reply:
yes